pkg/cmd/controller/start.go

```diff
@@ -24,103 +24,82 @@ import (
-	EtcdCoreCAConfigMapName string
-	EtcdCoreCertSecretName  string
-
-	EtcdProxyImage string
-
-	KubeconfigPath string
-
-	ControllerNamespace string
-}
-
 func NewCommandEtcdProxyControllerStart(stopCh <-chan struct{}) *cobra.Command {
-	o := &EtcdProxyControllerOptions{}
+	o := options.NewEtcdProxyOptions()
 
 	cmd := &cobra.Command{
 		Short: "Start EtcdProxyController",
 		Long:  "Start EtcdProxyController",
 		RunE: func(c *cobra.Command, args []string) error {
-			if err := o.RunController(stopCh); err != nil {
+			cfg, err := o.Config()
+			if err != nil {
+				return err
+			}
+
+			if err := RunController(cfg, stopCh); err != nil {
 				return err
 			}
 			return nil
 		},
 	}
 
-	flags := cmd.Flags()
-	flags.StringVar(&o.EtcdCoreURL, "core-etcd-url", "", "The address of the core etcd server.")
-	cmd.MarkFlagRequired("core-etcd-url")
-
-	flags.StringVar(&o.KubeconfigPath, "kubeconfig", "", "Path to kubeconfig (required only if running out-cluster).")
-
-	flags.StringVar(&o.EtcdCoreCAConfigMapName, "core-etcd-ca-configmap-name", "etcd-coreserving-ca", "The name of the ConfigMap where CA is stored.")
-	flags.StringVar(&o.EtcdCoreCertSecretName, "core-etcd-cert-secret-name", "etcd-coreserving-cert", "The name of the Secret where client certificates are stored.")
-	flags.StringVar(&o.EtcdProxyImage, "proxy-etcd-image", "quay.io/coreos/etcd:v3.2.18", "The image to be used for creating etcd proxy pods.")
+	o.AddFlags(cmd.Flags())
 	return cmd
 }
 
-// initControllerClientSets returns kubernetes clientset and etcdproxy clientset.
-func (o EtcdProxyControllerOptions) initControllerClientSets() (*kubernetes.Clientset, *clientset.Clientset, error) {
-	cfg, err := clientcmd.BuildConfigFromFlags("", o.KubeconfigPath)
-	if err != nil {
-		return nil, nil, err
-	}
-
-	kubeClient, err := kubernetes.NewForConfig(cfg)
-	if err != nil {
-		return nil, nil, err
-	}
-
-	etcdproxyClient, err := clientset.NewForConfig(cfg)
-	if err != nil {
-		return nil, nil, err
-	}
-
-	return kubeClient, etcdproxyClient, nil
+// initKubeClient returns Kubernetes Clientset.
+func initControllerClientSet(kubeconfig *restclient.Config) (*kubernetes.Clientset, error) {
+	return kubernetes.NewForConfig(kubeconfig)
 }
 
-// initControllerInformers returns informer namespaced to controller's namespace and etcdproxy informer.
-func (o EtcdProxyControllerOptions) initControllerInformers(kubeClientset *kubernetes.Clientset, etcdproxyClientset *clientset.Clientset,
-	defaultResync time.Duration) (kubeinformers.SharedInformerFactory, informers.SharedInformerFactory) {
-	kubeInformersNamespaced := kubeinformers.NewFilteredSharedInformerFactory(kubeClientset, defaultResync, o.ControllerNamespace, nil)
-	etcdproxyInformers := informers.NewSharedInformerFactory(etcdproxyClientset, defaultResync)
+// initEtcdProxyClient returns the EtcdProxy Clientset.
+func initEtcdProxyClient(kubeconfig *restclient.Config) (*clientset.Clientset, error) {
+	return clientset.NewForConfig(kubeconfig)
```

Comment:
do you need those funcs?

Reply:
No. Earlier, the logic was more complex. We can move code from those function directly to the `RunController` function.